compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -594,41 +506,167 @@ public STNode resumeParsing(ParserRuleContext context, Object... args) {
+            case EXPRESSION_RHS:
+                return parseExpressionRhs((OperatorPrecedence) args[0], (STNode) args[1], (boolean) args[2],
+                        (boolean) args[3], (boolean) args[4], (boolean) args[5]);
+            case CONSTANT_EXPRESSION_START:
+                return parseSimpleConstExprInternal();
+            case BINDING_PATTERN_OR_EXPR_RHS:
+                return parseTypedBindingPatternOrExprRhs((STNode) args[0], (boolean) args[1]);
+            case TYPE_DESC_OR_EXPR_RHS:
+                return parseTypeDescOrExprRhs((STNode) args[0]);
+            case WAIT_FUTURE_EXPR_END:
+                return parseWaitFutureExprEnd((int) args[0]);
+            case MEMBER_ACCESS_KEY_EXPR_END:
+                return parseMemberAccessKeyExprEnd();
+            case BRACED_EXPR_OR_ANON_FUNC_PARAM_RHS:
+                return parseBracedExprOrAnonFuncParamRhs((STNode) args[0], (STNode) args[1], (boolean) args[2]);
+            case EXPR_FUNC_BODY_START:
+                return parseDoubleRightArrow();
+            case REMOTE_CALL_OR_ASYNC_SEND_RHS:
+                return parseRemoteCallOrAsyncSendActionRhs((STNode) args[0], (boolean) args[1], (STNode) args[2]);
+            default:
+                return resumeKeywordParsing(context, args);
+        }
+    }
+
+    public STNode resumeStatementNodeParsing(ParserRuleContext context, Object... args) {
+        switch (context) {
+            case STATEMENT:
+                return parseStatement();
+            case STATEMENT_WITHOUT_ANNOTS:
+                return parseStatement((STNode) args[0]);
+            case STMT_START_WITH_EXPR_RHS:
+                return parseStatementStartWithExprRhs((STNode) args[0]);
+            case STMT_START_BRACKETED_LIST_MEMBER:
+                return parseStatementStartBracketedListMember();
+            case VAR_DECL_STMT_RHS:
+                return parseVarDeclRhs((STNode) args[0], (STNode) args[1], (STNode) args[2], (boolean) args[3]);
+            default:
+                return resumeActionOrExpressionNodesParsing(context, args);
+        }
+    }
+
+    public STNode resumeTopLevelNodeParsing(ParserRuleContext context, Object... args) {
+        switch (context) {
+            case TOP_LEVEL_NODE:
+                return parseTopLevelNode();
+            case TOP_LEVEL_NODE_WITHOUT_METADATA:
+                return parseTopLevelNode((STNode) args[0]);
+            case TOP_LEVEL_NODE_WITHOUT_MODIFIER:
+                return parseTopLevelNode((STNode) args[0], (STNode) args[1]);
+            case IMPORT_DECL_RHS:
+                return parseImportDecl((STNode) args[0], (STNode) args[1]);
+            case IMPORT_PREFIX:
+                return parseImportPrefix();
+            case IMPORT_MODULE_NAME:
+            case IMPORT_ORG_OR_MODULE_NAME:
+            case SERVICE_NAME:
+            case MODULE_ENUM_NAME:
+                return parseIdentifier(context);
+            case IMPORT_KEYWORD:
```

Comment:
Since theres a separate method for resuming keywords, shall we move all keywords in to that method?

Fix in all places